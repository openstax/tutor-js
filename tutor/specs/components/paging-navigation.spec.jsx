import keymaster from 'keymaster';
import Nav from '../../src/components/paging-navigation';

jest.mock('keymaster');

function TestComponent() {
    return <h1>Imma child</h1>;
}

describe('Paging Navigation', () => {
    let props;

    beforeEach(() => {
        props = {
            documentImpl: {
                title: 'test',
            },
            className: 'my-nav-test',
            onForwardNavigation:  jest.fn(),
            onBackwardNavigation: jest.fn(),
            isForwardEnabled:     true,
            isBackwardEnabled:    true,
            titles: {
                next: 'The Page That Comes After This One',
                current: 'Set From Nav',
                previous: 'The Page Before This One',
            },
        };
    });

    it('binds / unbinds on mount/unmount', () => {
        jest.useFakeTimers();
        const nav = shallow(<Nav {...props}><TestComponent /></Nav>);
        jest.runAllTimers();
        expect(keymaster).toHaveBeenCalledTimes(2);
        expect(keymaster).toHaveBeenCalledWith('left', expect.any(Function));
        expect(keymaster).toHaveBeenCalledWith('right', expect.any(Function));

        nav.setProps({ enableKeys: false });
        expect(keymaster.unbind).toHaveBeenCalledWith('left');
        expect(keymaster.unbind).toHaveBeenCalledWith('right');
        nav.setProps({ enableKeys: true });
        jest.runAllTimers();
        expect(keymaster).toHaveBeenCalledTimes(4);
        nav.unmount();
        expect(keymaster.unbind).toHaveBeenCalledTimes(4);
    });

    it('renders and matches snapshot', () => {
        expect.snapshot(<Nav {...props}><TestComponent /></Nav>).toMatchSnapshot();
    });

    it('is accessibile', async () => {
        jest.useRealTimers();
        const wrapper = mount(<Nav {...props}><TestComponent /></Nav>);
        wrapper.unmount()
    });

    it('sets titles', () => {
        const wrapper = shallow(<Nav {...props}><TestComponent /></Nav>);
        expect(props.documentImpl.title).toEqual('Set From Nav');
        expect(wrapper).toHaveRendered(`a.next[title="${props.titles.next}"]`);
        expect(wrapper).toHaveRendered(`a.prev[title="${props.titles.previous}"]`);
        wrapper.unmount()
    });

    it('calls prop fns', () => {
        const nav = shallow(<Nav {...props}><TestComponent /></Nav>);
        const preventDefault = jest.fn();
        nav.find('a.next').simulate('click', { preventDefault });
        expect(props.onForwardNavigation).toHaveBeenCalled();
        nav.find('a.prev').simulate('click', { preventDefault } );
        expect(props.onBackwardNavigation).toHaveBeenCalled();
        expect(preventDefault).toHaveBeenCalledTimes(2);
        nav.unmount()
    });

});
