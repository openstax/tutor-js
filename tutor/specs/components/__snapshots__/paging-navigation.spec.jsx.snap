// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging Navigation renders and matches snapshot 1`] = `
.c1 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.arrow-wrapper .c2.c2.c2.icon {
  top: calc(55vh - 80px);
  margin: 80px auto;
  height: 80px;
  width: 80px;
  border: 1px solid #5e6062;
  border-radius: 50%;
  padding: 20px 24px;
  opacity: 0.6;
}

.arrow-wrapper .c2.c2.c2.icon path {
  fill: #5e6062;
}

.c0.c0.arrow-wrapper {
  width: 100%;
  height: 100%;
  max-width: 100px;
}

<div
  className="tutor-paging-navigation my-nav-test"
>
  <a
    aria-controls="paged-content"
    className="paging-control prev"
    data-test-id="go-backward"
    disabled={false}
    href="#"
    onClick={[Function]}
    tabIndex={0}
    target="_blank"
    title="The Page Before This One"
  >
    <div
      className="c0 arrow-wrapper"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10 c1 ox-icon ox-icon-chevron-left c2 icon"
        data-icon="chevron-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </a>
  <div
    aria-live="polite"
    className="paged-content"
    id="paged-content"
    role="region"
    tabIndex="0"
  >
    <h1>
      Imma child
    </h1>
  </div>
  <a
    aria-controls="paged-content"
    className="paging-control next"
    data-test-id="go-forward"
    disabled={false}
    href="#"
    onClick={[Function]}
    tabIndex={0}
    title="The Page That Comes After This One"
  >
    <div
      className="c0 arrow-wrapper"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-right fa-w-10 c1 ox-icon ox-icon-chevron-right c2 icon"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </a>
</div>
`;
